### DISCUSSION TITLE
Laravel's validation library

### DISCUSSION THREAD

**Discussion by emreakay** [spatie/laravel-data#4] [Ideas]
Hi

Thank you for this great project, i have been using your dto package before but i will be using it from now on.

I have 2 question about the package.

- can we use laravel's Validator Facedes as php8 attribute?

i want to use this package to validate api requests and responses.
can we create different validators for same data object for create and update api requests 

Thanks.

**Comment by rubenvanassche** [comment]
At the moment with PHP 8 that would not be possible since we cannot nest attributes in any way. There is a possibility with PHP 8.1, you could then create two attributes:

- UpdateRules
- CreateRules

These two attributes then would extend from `ValidationAttribute` and take other attributes as input. When validating such an attribute you check if the request is update or create and would return the correct rules from the attributes given to the base `UpdateRules` or `CreateRules` attribute.

For now I would use the `rules` method on the data class and dynamically determine which rules should be used.

**Reply by TeaDrinkingProgrammer** [reply]
I am converting my project to Data classes and I am wondering about this exact problem: I currently have two requests: the update request which requires ids and the store request which does not. This sounds like a good solution, but as I am a beginner I cannot quite figure it out. As PHP 8.1 is out for a while now, could you maybe provide a snippet or, if this problem is as common as I would figure, dedicate a page for it on the documentation?
